### ISSUE TITLE
live capture not supported on Darwin

### ISSUE THREAD

**Issue by a1exus** [draios/sysdig#577]
I followed [How to Install Sysdig for Windows and OSX · draios/sysdig Wiki] (https://github.com/draios/sysdig/wiki/How-to-Install-Sysdig-for-Windows-and-OSX) and I got following error

> live capture not supported on Darwin

I'm using OS X El Capitan (10.11.4).

Thanks!

**Comment by aleks-f** [comment]
As the message says, live capture is not supported on Darwin. It is only supported on Linux:

> Sysdig is open source, system-level exploration: capture system state and activity from a running Linux instance.

What you can do on Darwin is replay the capture files recorded on Linux.

**Comment by jamesbrink** [comment]
I know this question is closed now, but is there any effort for this in progress or a roadmap?

**Comment by aleks-f** [comment]
There are no immediate plans for it. But sysdig is open source and we gladly accept contributions ;)

**Comment by etown** [comment]
Would it be feasible to convert a ktrace file to a sysdig capture format?

**Comment by jamesbrink** [comment]
okay great, I am not much of a c dev, but I would not mind making an attempt. I looked at the source code a bit, I dont yet know the best way to handle remote capture. I know i need to think about security, how to handle the buffer of data on capturing system. I assume sysdig will drop data it can't keep up with, similar to how tcpdump does? I have only been using it a few days thus far, I have a lot more I need to dig into. As far as getting started, beyond looking at the code are there any other resources available?

I did see this in the wiki
https://github.com/draios/sysdig/blob/master/coding_conventions.md

as well as the mailing list and irc channel. 

Thanks for the quick response @aleks-f

**Comment by aleks-f** [comment]
@etown: I guess it could be done, but would there be any real benefit compared to using kdump directly? csysdig, perhaps. However, as of OSX 10.5, ktrace has been replaced with dtrace

Which brings us to @jamesbrink's question - there are two parts to enabling live capture on mac:
(1) get syscalls/data out of the kernel space into the rignbuffer
(2) storing them into capture file. 

In regards to (1), there are 2 options:
(a) writing Darwin kernel module similar to [the Linux one](https://github.com/draios/sysdig/tree/dev/driver) 
(b) using [libdtrace](http://opensource.apple.com//source/xnu/xnu-3248.20.55/bsd/sys/dtrace.h)

The (b) option may be an easier path, but with a bump - libdtrace is private interface, so keeping things in sync across OS versions and flavors could be a busy task. A Good Thing about dtrace-based implementation is that it would lend itself nicely to BSD and Solaris/Illumos. FWIW, there's also [dtrace4linux](https://github.com/dtrace4linux/linux).

Regarding (2), the sysdig userspace code currently not enabled on OSX must be ported - there's some effort there, no doubt.

Bottom line - it is not a small or simple task, but don't let that discourage you ;)

Explanation and comparison between sysdig, strace and dtrace can be found [here](https://sysdig.com/blog/sysdig-vs-dtrace-vs-strace-a-technical-discussion/).

**Comment by etown** [comment]
@aleks-f thanks that's really good explanation.
Yes, I think the advantage would be csysdig and the ability to run chisels.

I was proposing the (b) path because I figured it would be easier, since I don't no how tracepoint functionality works within the darwin kernel extension framework.
